docs/samples/Microsoft.ML.Samples/Dynamic/Transforms/Text/LatentDirichletAllocation.cs

```diff
@@ -0,0 +1,74 @@
+﻿using System;
+using System.Collections.Generic;
+using Microsoft.ML.Data;
+
+namespace Microsoft.ML.Samples.Dynamic
+{
+    public static class LatentDirichletAllocation
+    {
+        public static void Example()
+        {
+            // Create a new ML context, for ML.NET operations. It can be used for exception tracking and logging, 
+            // as well as the source of randomness.
+            var mlContext = new MLContext();
+
+            // Create a small dataset as an IEnumerable.
+            var samples = new List<TextData>()
+            {
+                new TextData(){ Text = "ML.NET's LatentDirichletAllocation API computes topic model." },
+                new TextData(){ Text = "ML.NET's LatentDirichletAllocation API is the best for topic model." },
+                new TextData(){ Text = "I like to eat broccoli and banana." },
+                new TextData(){ Text = "I eat a banana in the breakfast." },
+                new TextData(){ Text = "This car is expensive compared to last week's price." },
+                new TextData(){ Text = "This car was $X last week." },
+            };
+
+            // Convert training data to IDataView.
+            var dataview = mlContext.Data.LoadFromEnumerable(samples);
+
+            // A pipeline for featurizing the text/string using LatentDirichletAllocation API.
+            // To be more accurate in computing the LDA features, the pipeline first normalizes text and removes stop words
+            // before passing tokens to LatentDirichletAllocation.
+            var pipeline = mlContext.Transforms.Text.NormalizeText("normText", "Text")
+                .Append(mlContext.Transforms.Text.TokenizeIntoWords("Tokens", "normText"))
+                .Append(mlContext.Transforms.Text.RemoveDefaultStopWords("Tokens"))
+                .Append(mlContext.Transforms.Conversion.MapValueToKey("Tokens"))
+                .Append(mlContext.Transforms.Text.ProduceNgrams("Tokens"))
+                .Append(mlContext.Transforms.Text.LatentDirichletAllocation("Features", "Tokens", numberOfTopics: 3));
+
+            // Fit to data.
+            var transformer = pipeline.Fit(dataview);
+
+            // Create the prediction engine to get the LDA features extracted from the text.
+            var predictionEngine = mlContext.Model.CreatePredictionEngine<TextData, TransformedTextData>(transformer);
```

Comment:
>predictionEngine [](start = 16, length = 16)

Similar to the other PR, I wonder if we should stay entirely within IDataView and not create a prediction engine. That is, use a `TakeRows` filter followed by a `CreateEnumerable`.

Reply:
This is done because some of the transforms related to text processing such as (NormalizeText. TokenizeIntoWords etc.) don't need training data. In such cases, prediction engine seems more appropriate. But we can definitely have consensus on this. I will follow up.

---
In reply to: [271981536](https://github.com/dotnet/machinelearning/pull/3191#discussion_r271981536) [](ancestors = 271981536)